app/streaming/video/ffmpeg-renderers/d3d11va.cpp

```diff
@@ -150,11 +174,130 @@ D3D11VARenderer::~D3D11VARenderer()
+            );
+    }
+}
+
+/**
+ * \brief Set Monitor HDR MetaData information
+ *
+ * Get the Monitor HDT MetaData via LimeLight library
+ *
+ * \param PSS_HDR_METADATA* HDRMetaData The varaible to set the metadata information
+ * \return bool Return True is succeed
+ */
+void D3D11VARenderer::setHDROutPut(){
+
+    DXGI_HDR_METADATA_HDR10 outputHDRMetaData;
+
+    // Find the monitor attached to the application
+    Microsoft::WRL::ComPtr<IDXGIOutput> pOutput;
+    if (SUCCEEDED(m_SwapChain->GetContainingOutput(&pOutput))) {
+        Microsoft::WRL::ComPtr<IDXGIOutput6> pOutput6;
+        if (SUCCEEDED(pOutput.As(&pOutput6))){
+            DXGI_OUTPUT_DESC1 desc1 {};
+            if (SUCCEEDED(pOutput6->GetDesc1(&desc1))){
+                // Get Monitor ColorSpace for SDR and HDR (if the monitor is capable of HDR)
+                m_OutputColorSpace = desc1.ColorSpace;
+
+                // Magic constants to convert to fixed point.
+                // https://docs.microsoft.com/en-us/windows/win32/api/dxgi1_5/ns-dxgi1_5-dxgi_hdr_metadata_hdr10
+                static constexpr int kPrimariesFixedPoint = 50000;
+                static constexpr int kMinLuminanceFixedPoint = 10000;
+
+                // Format Monitor HDR MetaData
+                outputHDRMetaData.RedPrimary[0] = desc1.RedPrimary[0] * kPrimariesFixedPoint;
+                outputHDRMetaData.RedPrimary[1] = desc1.RedPrimary[1] * kPrimariesFixedPoint;
+                outputHDRMetaData.GreenPrimary[0] = desc1.GreenPrimary[0] * kPrimariesFixedPoint;
+                outputHDRMetaData.GreenPrimary[1] = desc1.GreenPrimary[1] * kPrimariesFixedPoint;
+                outputHDRMetaData.BluePrimary[0] = desc1.BluePrimary[0] * kPrimariesFixedPoint;
+                outputHDRMetaData.BluePrimary[1] = desc1.BluePrimary[1] * kPrimariesFixedPoint;
+                outputHDRMetaData.WhitePoint[0] = desc1.WhitePoint[0] * kPrimariesFixedPoint;
+                outputHDRMetaData.WhitePoint[1] = desc1.WhitePoint[1] * kPrimariesFixedPoint;
+                outputHDRMetaData.MaxMasteringLuminance = desc1.MaxLuminance;
+                outputHDRMetaData.MinMasteringLuminance = desc1.MinLuminance * kMinLuminanceFixedPoint;
+                // MaxContentLightLevel is not available in DXGI_OUTPUT_DESC1 structure
+                // https://learn.microsoft.com/fr-fr/windows/win32/api/dxgi1_6/ns-dxgi1_6-dxgi_output_desc1
+                // But MaxContentLightLevel is not needed and greater or equal to MaxFullFrameLuminance, so it is safe to set a minimum for it
+                // https://professionalsupport.dolby.com/s/article/Calculation-of-MaxFALL-and-MaxCLL-metadata
+                // Also note that these are not fixed-point.
+                outputHDRMetaData.MaxContentLightLevel = desc1.MaxFullFrameLuminance;
+                outputHDRMetaData.MaxFrameAverageLightLevel = desc1.MaxFullFrameLuminance;
+            }
+        }
+    }
+
+    if (m_VideoProcessor.p) {
```

Comment:
I think you should be able to check for null with these `CComPtr` objects by simply using `if (m_VideoProcessor)` rather than directly accessing the internal pointer.

Reply:
Done